addons/account/models/account_journal_dashboard.py

```diff
@@ -677,19 +677,8 @@ def open_action(self):
         action['context'] = context
         action['context'].update({
             'default_journal_id': self.id,
-            'search_default_journal_id': self.id,
```

Comment:
That's right, but to me, you need to explicitly set action['domain'] = [('journal_id', '=', self.id)].

Without that, clicking misc journals on the dashboard opens all moves from other journals as well (invoice and bill journals seem to work differently ; you might want to check why it works for them ; we could be missing something important)

Reply:
After double checking the behavior
When opening invoices we use `action_move_out_invoice_type` which has a domain defined 
`<field name="domain">[('move_type', '=', 'out_invoice')]</field>`
https://github.com/odoo/odoo/blob/c30c1dac33da05a4064d5353215098b2a5b3c87d/addons/account/views/account_move_views.xml#L1465
This is not the case when opening misc journals as we use `action_move_journal_line` with no domain
https://github.com/odoo/odoo/blob/c30c1dac33da05a4064d5353215098b2a5b3c87d/addons/account/views/account_move_views.xml#L1441